### PR TITLE
PP-5166 Add SQS configuration

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID

Adds SQS configuration and initialises SqsClient to interact with SQS

### CODE REVIEW

File: src/main/resources/config/config.yaml [hunk 1]
```diff
@@ -107,6 +107,11 @@ captureProcessConfig:
   maximumRetries: ${CAPTURE_PROCESS_MAXIMUM_RETRIES:-96}
   captureUsingSQS: ${CAPTURE_USING_SQS_FEATURE_FLAG:-false}
 
+sqsConfig:
+  endpoint: ${AWS_SQS_ENDPOINT}
+  captureQueueUrl: ${AWS_SQS_CAPTURE_QUEUE_ID}
```
Comment (kbottla) [inline src/main/resources/config/config.yaml]: Env variable is passed here https://github.com/alphagov/pay-infra/pull/2123/files#diff-15da3aa23a29dabb07b9753b510715d7L111

File: src/main/java/uk/gov/pay/connector/app/ConnectorModule.java [hunk 2]
```diff
@@ -120,4 +125,23 @@ protected NotifyClientFactory getNotifyClientFactory(ConnectorConfiguration conn
     public Queue<ChargeEntity> captureQueue(ConnectorConfiguration connectorConfiguration) {
         return new ArrayBlockingQueue(connectorConfiguration.getCaptureProcessConfig().getBatchSize());
     }
+
+    @Provides
+    @Singleton
+    public AmazonSQS sqsClient(ConnectorConfiguration connectorConfiguration) {
+
+        if (isEmpty(connectorConfiguration.getSqsConfig().getEndpoint())) {
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/app/ConnectorModule.java:133]: As explained in README.md, this is not required for AWS if `AWS_SQS_REGION` is set. But for a not standard SQS this is required

**Review by sfount — APPROVED** [review]

**Comment by sfount** [comment]
Configuration looks right, lines up with story. We can double check if we need any further configuration options for local Docker containers when that is merged in pay-infra

File: README.md [hunk 3]
```diff
@@ -42,6 +42,9 @@ The following variables control the background process:
 | `CAPTURE_PROCESS_BATCH_SIZE` | `10` | limits the batch window size processed at each polling attempt. If connector is not managing to clear the queue of captures, increase this value. |
 | `CAPTURE_PROCESS_RETRY_FAILURES_EVERY` | `60 minutes` | a failed capture attempt will be returned to the queue, and will not be retried until this time has passed |
 | `CAPTURE_PROCESS_MAXIMUM_RETRIES` | `48` | connector keeps track of the number of times capture has been attempted for each charge. If a charge fails this number of times or more it will be marked as a permanent failure. An error log message will be written as well. This should *never* happen and if it does it should be investigated. |
+| `AWS_SQS_ENDPOINT` | - |  URL that is the entry point for SQS. Not required when AWS_SQS_REGION is set or can be set if using a non-standard SQS endpoint   |
+| `AWS_SQS_CAPTURE_QUEUE_ID` | - | SQS capture queue URL  |
```
Comment (sfount) [inline README.md:46]: @kbottla as we noticed this afternoon - this README should be updated to `AWS_SQS_CAPTURE_QUEUE_URL`

**Pull request merged**